### PR TITLE
docs: Update contributor guide with installation instructions 

### PR DESCRIPTION
**Pull request by caicancai**
## Which issue does this PR close?

Add environment information

### CODE REVIEW

**Comment by caicancai** [comment]
The dnf command needs to be installed on Linux

File: docs/source/contributor-guide/index.md [hunk 1]
```diff
@@ -95,9 +95,13 @@ Compiling DataFusion from sources requires an installed version of the protobuf
 On most platforms this can be installed from your system's package manager
 
 ```
+#Linux
```
Comment (jackwener) [inline docs/source/contributor-guide/index.md:98]:
```suggestion
# Linux
```

File: docs/source/contributor-guide/index.md [hunk 2]
```diff
@@ -95,9 +95,13 @@ Compiling DataFusion from sources requires an installed version of the protobuf
...
 $ sudo apt install -y protobuf-compiler
+$ sudo apt install -y dnf
```
Comment (jackwener) [inline docs/source/contributor-guide/index.md:100]:
```suggestion
$ sudo apt install -y protobuf-compiler dnf
```

File: docs/source/contributor-guide/index.md [hunk 3]
```diff
@@ -95,9 +95,13 @@ Compiling DataFusion from sources requires an installed version of the protobuf
...
 $ dnf install -y protobuf-devel
 $ pacman -S protobuf
+
+#Macos
```
Comment (jackwener) [inline docs/source/contributor-guide/index.md:104]:
```suggestion
# macOS
```

**Comment by caicancai** [comment]
Thank you for your review

File: docs/source/contributor-guide/index.md [hunk 4]
```diff
@@ -95,9 +95,12 @@ Compiling DataFusion from sources requires an installed version of the protobuf
 On most platforms this can be installed from your system's package manager
 
 ```
-$ sudo apt install -y protobuf-compiler
+# Linux
+$ sudo apt install -y protobuf-compiler dnf
 $ dnf install -y protobuf-devel
 $ pacman -S protobuf
+
+# macOS
```
Comment (waynexia) [inline docs/source/contributor-guide/index.md:103]: 👍

**Review by waynexia — COMMENTED** [review]
Can you give a brief explanation of the rationale? `dnf` is good but it seems not a required step to set up DataFusion's environment. The origin three (or four, since macOS is separated) commands are for different systems.

**Comment by waynexia** [comment]
Or we can do the same like macOS:

```
# Ubuntu
$ sudo apt install -y protobuf-compiler

# Fedora
$ dnf install -y protobuf-devel

# Arch Linux
$ pacman -S protobuf
```

File: docs/source/contributor-guide/index.md (hunk 4 above)
Comment (caicancai) [reply docs/source/contributor-guide/index.md:103]: I don’t know, but when I use the dnf command according to the documentation, it shows that I have not installed the dnf command.
In fact, I didn’t install the dnf command to execute cargo build and there was no problem.

**Comment by caicancai** [comment]
> Can you give a brief explanation of the rationale? `dnf` is good but it seems not a required step to set up DataFusion's environment. The origin three (or four, since macOS is separated) commands are for different systems.



> Can you give a brief explanation of the rationale? `dnf` is good but it seems not a required step to set up DataFusion's environment. The origin three (or four, since macOS is separated) commands are for different systems.

I don’t know, but when I use the dnf command according to the documentation, it shows that I have not installed the dnf command.
In fact, I didn’t install the dnf command to execute cargo build and there was no problem.

**Comment by waynexia** [comment]
> I don’t know, but when I use the dnf command according to the documentation, it shows that I have not installed the dnf command.

Ahh, I get it. Those commands are meant to be provided separately for different environments lol

**Comment by caicancai** [comment]
> Or we can do the same like macOS:
> 
> ```
> # Ubuntu
> $ sudo apt install -y protobuf-compiler
> 
> # Fedora
> $ dnf install -y protobuf-devel
> 
> # Arch Linux
> $ pacman -S protobuf
> ```

it looks like this is better

**Comment by caicancai** [comment]
> > I don’t know, but when I use the dnf command according to the documentation, it shows that I have not installed the dnf command.
> 
> Ahh, I get it. Those commands are meant to be provided separately for different environments lol

That's it, thank you for your answer

**Review by jackwener — APPROVED** [review]
Thanks @caicancai @waynexia

**Review by waynexia — APPROVED** [review]

**Comment by caicancai** [comment]
Thank you for your suggestions and reviews

**Pull request merged**